### PR TITLE
refactor: Fix calculating balance on particular block

### PR DESCRIPTION
**Pull request by shelegdmitriy**

### CODE REVIEW

File: src/lib/lockup.ts [hunk 1]
```diff
@@ -96,6 +107,42 @@ export const lookupLockup = async (
...
+ * @param accountId near lockup accountId used to interact with the network.
+ * @param nearConfig specify custom connection to NEAR network.
+ * @param blockReference specify block of calculated data.
+ * @returns lockup account information {@link ViewAccount}
+ */
+export const viewLockupAccountBalance = async (
+  accountId: string,
+  nearConfig?: ConnectConfig,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  blockReference: BlockReference | any = { finality: "final" }
```
Comment (frol) [inline src/lib/lockup.ts]: Why `any` is here?

File: src/tests/lockup.spec.ts [hunk 2]
```diff
@@ -20,10 +21,32 @@ test("view all info about lockup account on 'finality' block", async (t) => {
...
+    nodeUrl: "https://archival-rpc.mainnet.near.org",
+    networkId: "mainnet",
+    keyStore: connectOptions.keyStore,
+    headers: connectOptions.headers
+  };
+
   const res = await viewLockupAccount(
     lockupDataMock[1].lockupAccountId,
-    null,
+    arhivalConfig,
```
Comment (frol) [inline src/tests/lockup.spec.ts]: Now this and the next test are almost identical. I suggest merge them and have a loop over the "lockupDataMock" array (see other comment about "lockupDataMock")

File: src/tests/lockup.spec.ts [hunk 3]
```diff
@@ -20,10 +21,32 @@ test("view all info about lockup account on 'finality' block", async (t) => {
...
+
+test("view all info about account on particular block using arhival credencials", async (t) => {
+  const arhivalConfig = {
+    nodeUrl: "https://archival-rpc.mainnet.near.org",
+    networkId: "mainnet",
+    keyStore: connectOptions.keyStore,
+    headers: connectOptions.headers
+  };
+  const res = await viewLockupAccount(
+    lockupDataMock[2].lockupAccountId,
```
Comment (frol) [inline src/tests/lockup.spec.ts]: > mock objects are simulated objects that mimic the behavior of real objects in controlled ways

What you have here is not a mock, it is a snapshot. I suggest we use `lockupAccountSnapshots` name here.

File: src/tests/mock.ts [hunk 4]
```diff
@@ -63,5 +65,35 @@ export const lockupDataMock: readonly AccountLockup[] = [
...
+      new BN("0").toString(),
+      2
+    ),
+    liquidAmount: nearAPI.utils.format.formatNearAmount(
+      new BN("35000647738262344800000000")
+      .sub(new BN("0"))
+      .toString(),
+      2
+    ),
+    totalAmount: "1,900,035",
```
Comment (frol) [inline src/tests/mock.ts]: I would keep them in BN type rather than exposing formatted strings

File: src/types/types.ts [hunk 5]
```diff
@@ -62,3 +63,21 @@ export type ViewStateResult = QueryResponseKind & {
   readonly values: readonly StateItem[];
   readonly proof: readonly string[];
 };
+
+export type ViewAccountQuery = QueryResponseKind & {
+  readonly amount: string;
+  readonly locked: string;
+  readonly code_hash: string;
+  readonly storage_usage: number;
+  readonly storage_paid_at: number; //block_height
```
Comment (frol) [inline src/types/types.ts]: Just ignore it (remove it). This field is soft-deprecated on nearcore side

File: src/types/types.ts [hunk 6]
```diff
@@ -62,3 +63,21 @@ export type ViewStateResult = QueryResponseKind & {
...
+};
+
+export type ViewAccount = {
+  readonly amount: string;
+  readonly locked: string;
+  readonly codeHash: string;
+  readonly storageUsage: number;
+  readonly storagePaidAt: number; //block_height
```
Comment (frol) [inline src/types/types.ts]: Just ignore it (remove it). This field is soft-deprecated on nearcore side

**Review by frol — APPROVED** [review]
LGTM overall, but as per the offline discussion, let's add a few more snapshots by taking the data at the moments before unlock is started, at the first block when it started, at the block before the final unlock, and the block when all the tokens are unlocked.

**Comment by shelegdmitriy** [comment]
@frol I guess we can move on with this snapshots and tests if you don't mind

**Review by frol — APPROVED** [review]

**Pull request merged**